app/services/api_keys/fetch_service.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module ApiKeys
+  class FetchService < BaseService
+    Result = BaseResult[:api_key, :organization]
+
+    def initialize(auth_token)
+      @auth_token = auth_token
+      super
+    end
+
+    def call
+      api_key_json = Rails.cache.read("api_key/#{auth_token}")
+      if api_key_json
```

Comment:
perhaps it makes sense to double check the expiration here? I know it technically would not be needed, but it's an easy sanity check to add.